File: flutter_project/lib/models/food_item.dart
class FoodItem {
  final int id;
  final int vendorId;
  final String name;
  final double price;
  final String category;
  final String imageUrl;
  final String description;
  final bool isAvailable;
  final int calories;
  final String allergens;

  FoodItem({
    required this.id,
    required this.vendorId,
    required this.name,
    required this.price,
    required this.category,
    required this.imageUrl,
    required this.description,
    this.isAvailable = true,
    this.calories = 250,
    this.allergens = 'None',
  });

  factory FoodItem.fromJson(Map<String, dynamic> json) {
    return FoodItem(
      id: json['id'] ?? 0,
      vendorId: json['vendor_id'] ?? json['vendorId'] ?? 0,
      name: json['name'] ?? '',
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
      category: json['category'] ?? 'General',
      imageUrl: json['image_url'] ?? json['imageUrl'] ?? '',
      description: json['description'] ?? '',
      isAvailable: json['is_available'] == 1 || json['is_available'] == true || json['isAvailable'] == true,
      calories: json['calories'] ?? 250,
      allergens: json['allergens'] ?? 'None',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vendor_id': vendorId,
      'name': name,
      'price': price,
      'category': category,
      'image_url': imageUrl,
      'description': description,
      'is_available': isAvailable,
      'calories': calories,
      'allergens': allergens,
    };
  }
}

File: backend/app/Models/FoodItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class FoodItem extends Model
{
    use HasFactory;

    protected $table = 'food_items';

    protected $fillable = [
        'vendor_id',
        'name',
        'price',
        'category',
        'image_url',
        'description',
        'is_available',
        'initial_stock',
        'current_stock',
        'low_stock_threshold', 'sku', 'preparation_minutes', 'dietary_tags', 'allergen_info', 'is_featured',
    ];

    protected $casts = [
        'price' => 'double',
        'is_available' => 'boolean',
        'initial_stock' => 'integer',
        'current_stock' => 'integer',
        'low_stock_threshold' => 'integer',
        'preparation_minutes' => 'integer', 'dietary_tags' => 'array', 'is_featured' => 'boolean',
    ];

    /**
     * Graceful fallback accessor for initial_stock
     */
    public function getInitialStockAttribute($value)
    {
        return $value !== null ? (int) $value : 50;
    }

    /**
     * Graceful fallback accessor for low_stock_threshold
     */
    public function getLowStockThresholdAttribute($value)
    {
        return $value !== null ? (int) $value : 10;
    }

    /**
     * Get the Vendor User that offers this food dish.
     */
    public function vendor()
    {
        return $this->belongsTo(User::class, 'vendor_id');
    }

    /**
     * Orders associated with this food dish.
     */
    public function orders()
    {
        return $this->hasMany(Order::class, 'food_item_id');
    }
}
